invenio_app_ils/circulation/mail/tasks.py

```diff
@@ -12,59 +12,74 @@
 from flask import current_app
 from invenio_circulation.proxies import current_circulation
 
-from invenio_app_ils.circulation.mail.factory import loan_message_factory
+from invenio_app_ils.circulation.mail.factory import \
+    loan_message_creator_factory
+from invenio_app_ils.circulation.search import get_all_expiring_loans, \
+    get_all_overdue_loans
 from invenio_app_ils.circulation.utils import circulation_overdue_loan_days
-from invenio_app_ils.documents.api import Document
+from invenio_app_ils.mail.messages import get_common_message_ctx
 from invenio_app_ils.mail.tasks import send_ils_email
-from invenio_app_ils.records.api import Patron
 
 celery_logger = get_task_logger(__name__)
 
 
-def send_loan_mail(trigger, loan, message_ctx={}, **kwargs):
+def send_loan_mail(loan_action, loan, message_ctx={}, **kwargs):
```

Comment:
minor: I think it is more of a `reason` to send an email rather than `loan_action`. Also loans cannot act.

Reply:
Not sure I understand, but actually I might have reverted by mistake, it was `action`: this is because an action was performed on a loan, therefore -> send an email